src/bootstrap/bootstrap.py

```diff
@@ -643,18 +643,33 @@ def get_answer():
             if ostype != "Linux":
                 return False
 
-            # If the user has asked binaries to be patched for Nix, then
-            # don't check for NixOS.
+            # If the user has explicitly indicated whether binaries should be
+            # patched for Nix, then don't check for NixOS.
             if self.get_toml("patch-binaries-for-nix", "build") == "true":
                 return True
+            if self.get_toml("patch-binaries-for-nix", "build") == "false":
+                return False
+
+            # Assume we should fix until we see evidence that it is not NixOS
+            should_fix_retval = True
```

Comment:
This value always gets overridden below because the `try` does not catch other types of exceptions aside from `FileNotFoundError`. So it doesn't actually default to `True`

Reply:
I do not see the problem: if `/etc/os-release` does exist, but does not contain the NixOS signature, then it gets set to False, and I think that if the file doesn't exist, then this is a good enough evidence that this is not NixOS. The only remaining question is How does this behave on Windows, but I'm sure something sensible happens.

Reply:
I think you're right. 

I wrote this code originally prior to PR #115090, and in that in that context, there was a series of return statements that I was replacing here with a series of assignments and a return at the end. Then when I rebased over the effects of PR #115090, I blindly kept that form without really thinking about whether this value could ever be read.

Reply:
Fixed by ec2c95e0932312c994ee8c5d52b7ee93b5ae65b9